seed/challenges/02-javascript-algorithms-and-data-structures/functional-programming.json

```diff
@@ -40,9 +40,9 @@
         "Functional programming is a style of programming where solutions are simple, isolated functions, without any side effects outside of the function scope.",
         "<code>INPUT -> PROCESS -> OUTPUT</code>",
         "Functional programming is about:",
-        "1) Isolated functions - there is no dependence on external state, which includes global variables that are subject to change",
+        "1) Isolated functions - there is no dependence on the state of the program, which includes global variables that are subject to change",
         "2) Pure functions - the same input always gives the same output",
-        "3) Functions have no side effects - they do not mutate (change) external states",
+        "3) Functions have no side effects - they do not mutate, or change, the state of the program outside the function",
```

Comment:
Sadly, this is a very academic statement and in no way resembles the real world. Functions aim to have no side effects, and most should not. Some will necessarily have to, and they should be carefully controlled. Without some kind of side effects we can't do anything actually useful to the user in the end.